doc/developers/contributing.rst

```diff
@@ -1087,6 +1087,18 @@ When the change is in a class, we validate and raise warning in ``fit``::
   import warnings
 
   class ExampleEstimator:
+      """Changing the default value of an estimator parameter.
+
+      Parameters
+      ----------
+      n_clusters : int
+        Number of clusters.
+
+        .. versionchanged:: 0.20
```

Comment:
```suggestion
        .. versionchanged:: 0.22
```

Reply:
it seems more natural to put 0.22 but isn't it strange to put a number that is a version that does not exist yet?

Reply:
Thinking about it more, when there's just a change of default, I think we should just explain the change in plain text in the doc and only add the `versionchanged` once the change is effective. Otherwise it will be too easy to be out of sync with the indicated versions, and with the use of past/present/future

Reply:
@albertcthomas I agree that the version in `versionchanged` should be when the change will be effective.

@NicolasHug If I get it right, you expect that we go back in 0.22 to add the `versionchanged`. I don't see any safeguard avoiding to forget to make the change. It would actually be nice to have a programmatic way to detect if we miss it.

Reply:
I agree it's not foolproof. We should make it a habit with ourselves to write those entries when we make those "deprecation cleaning" PRs.

The issue with writing the entry right away is that it will be "the default will change in 0.22", but we'll most definitely forget to update that in 0.22 to use past tense. So in 0.23 and above, this will still be "default *will* change", and that'll bring confusion, which is IMHO worse than just not having the entry (we haven't had those entries until very recently).

Reply:
> If I get it right, you expect that we go back in 0.22 to add the `versionchanged`. I don't see any safeguard avoiding to forget to make the change. It would actually be nice to have a programmatic way to detect if we miss it.

I am going a bit out of the scope of this PR but I always saw this as a hard task to not miss or break anything when removing the `FutureWarning`, changing the `self._n_clusters` in `self.n_clusters` and removing `filterwarnings` in the tests. Especially because it is very often a different contributor that does this job after a release. Actually this is also a good incentive to think twice before deprecating something :)

For instance how can you easily find all the filterwarnings associated to a given deprecation? I remember some discussions about how to improve this.

Reply:
Another, not optimal solution, would be to use the present tense:
`The default value changes to 10 in version 0.22.`

Reply:
> For instance how can you easily find all the filterwarnings associated to a given deprecation? I remember some discussions about how to improve this.

When we review a deprecation we need to remember that basically everything needs to be tagged with a short "remove in 0.xx", bonus point if the reason can be given in a short amount of space

Reply:
Tagged in the warning message or in code comments? We might add this to the doc.

Reply:
> Tagged in the warning message or in code comments?

I think that we usually do both.